### PR TITLE
devilutionx: 1.4.1 -> 1.5.2

### PR DESCRIPTION
**Pull request by Zaechus**
## Description of changes

I had to use the upstream version of SDL_audiolib to get Diablo to run. I also removed the VERSION_NUM CMake flag from the derivation to get the devilutionX version to appear in-game. `devilutionx.mpq` also needed to be copied into the same directory as the binary in order to avoid the font issue from #238457.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/devilutionx/default.nix [hunk 1]
```diff
@@ -107,6 +107,7 @@ stdenv.mkDerivation rec {
     mv devilutionx.app $out/Applications
   '' else ''
     install -Dm755 -t $out/bin devilutionx
+    install -Dm755 -t $out/bin devilutionx.mpq
```
Comment (karolchmist) [inline pkgs/games/devilutionx/default.nix:110]: Is this needed for Darwin as well?

**Review by karolchmist — APPROVED** [review]
It works !

File: pkgs/games/devilutionx/default.nix [hunk 2]
```diff
@@ -21,65 +19,68 @@ let
...
+      --replace-fail "GIT_REPOSITORY https://github.com/diasurgical/libzt.git" "" \
+      --replace-fail "GIT_TAG ${libzt.rev}" "SOURCE_DIR ${libzt}"
+    substituteInPlace 3rdParty/SDL_audiolib/CMakeLists.txt --replace-fail "${SDL_audiolib.url}" "${SDL_audiolib}"
+    substituteInPlace 3rdParty/simpleini/CMakeLists.txt --replace-fail "${simpleini.url}" "${simpleini}"
   '';
 
   cmakeFlags = [
-    "-DBINARY_RELEASE=ON"
-    "-DVERSION_NUM=${version}"
+    "-DCMAKE_BUILD_TYPE=Release"
```
[the remark this answers is not in the document]
Comment (Zaechus) [reply pkgs/games/devilutionx/default.nix]: you're right, fixed.

File: pkgs/games/devilutionx/default.nix (hunk 1 above)
Comment (Zaechus) [reply pkgs/games/devilutionx/default.nix:106]: I'm unable to test, but I don't think so. The macos dmg file from their release page includes files like `24-00.clx` manually without including the devilutionx.mpq file. Also, their [CMake/Assets.cmake](https://github.com/diasurgical/devilutionX/blob/7ea712428ebf3df5da58e107cc40e8db3825bebc/CMake/Assets.cmake#L162) demonstrates that Apple is an exception to building the MPQ.

**Review by karolchmist — APPROVED** [review]

File: pkgs/games/devilutionx/default.nix [hunk 3]
```diff
@@ -21,67 +19,66 @@ let
...
     repo = "libzt";
     fetchSubmodules = true;
-    rev = "37a2efb0b925df632299ef07dc78c0af5f6b4756";
-    sha256 = "sha256-+o4ZTVqh4MDZES9m7mkfkMRlRDMBytDBuA0QIlnp73U=";
+    rev = "d6c6a069a5041a3e89594c447ced3f15d77618b8";
+    sha256 = "sha256-ttRJLfaGHzhS4jd8db7BNPWROCti3ZxuRouqsL/M5ew=";
+  };
+
+  # breaks without this version
+  SDL_audiolib = fetchurl {
```
Comment (aanderse) [inline pkgs/games/devilutionx/default.nix:48]: what about [this](https://github.com/NixOS/nixpkgs/commit/996bc09ba86b7d421bd93263edcd31b3f3ab0d71) instead

why? if any dependency of `devilutionx` changes then we need to recompile `SDL_audiolib` too... not a huge saving, but a saving nonetheless

Comment (Zaechus) [reply pkgs/games/devilutionx/default.nix]: added

**Comment by karolchmist** [comment]
Result of `nixpkgs-review pr 303308` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>devilutionx</li>
  </ul>
</details>

**Review by karolchmist — APPROVED** [review]

**Review by aanderse — APPROVED** [review]
thanks so much @Zaechus!

**Pull request merged**